pymodm/fields.py

```diff
@@ -1062,8 +1062,27 @@ def to_python(self, value):
             return self.related_model.from_document(value)
         return value
 
+    @staticmethod
+    def _value_to_mongo(related_model, value):
+        if isinstance(value, bson.SON):
+            return value
+
+        if isinstance(value, related_model):
+            return value.to_son()
+
+        # if value is a dict convert in to model
+        # so we can properly generate SON
+        if isinstance(value, dict):
+            return related_model.from_document(value).to_son()
+
+        # TODO: Should we raise a ValidationError as we could not
+        # convert value to SON?
+        return value
+
+
     def to_mongo(self, value):
-        return value.to_son()
+        return EmbeddedDocumentField._value_to_mongo(self.related_model,
+                                                     value)
```

Comment:
I'd prefer to move `_value_to_mongo` to [RelatedModelFieldsBase](https://github.com/mongodb/pymodm/blob/ce603f5079fb602eeb2421e1bca767f4cb2bd8ac/pymodm/base/fields.py#L194) as an instance/classmethod and call it something like `_model_to_document`, to make it clearer what it's doing. Then, we can call it here like `self._model_to_document(...)`.